dll/win32/shell32/iconcache.cpp

```diff
@@ -654,34 +654,15 @@ void SIC_Destroy(void)
  */
 static int SIC_LoadOverlayIcon(int icon_idx)
 {
-    WCHAR buffer[1024], wszIdx[8];
-    HKEY hKeyShellIcons;
-    LPCWSTR iconPath;
+    LPWSTR iconPath;
     int iconIdx;
 
     iconPath = swShell32Name;    /* default: load icon from shell32.dll */
     iconIdx = icon_idx;
 
-    if (RegOpenKeyExW(HKEY_LOCAL_MACHINE, L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Shell Icons",
-                      0, KEY_READ, &hKeyShellIcons) == ERROR_SUCCESS)
+    if (!HLM_GetIconW(icon_idx, iconPath, MAX_PATH, &iconIdx))
```

Comment:
The function does not overwrite the variable on failure, does it? It simply returns `FALSE` without any modification of output arguments if the key or value does not exist.

Reply:
And if the value exists and function succeeds?

Reply:
In that case, it indeed overwrites that, but it should load the correct icon based on the value.

Reply:
And then you will get undefined behavior in shell32, because `swShell32Name` is supposed to contain its path and not something else. Fixed in ca3804fd918.

Reply:
OK, thanks. My mistake then.